### DISCUSSION TITLE
Multiple Datas on a LCD Display

### DISCUSSION THREAD

**Discussion by 8chM** [jomjol/AI-on-the-edge-device#1873] [Q&A]
I´ve tried to implement a solution with your project to check my dosing system for pool, and publish this data on home assistant.
In my Display are three lines with data.
1st line pH:
2nd line Redox:
3rd line Temperature:

How I can implement that?
I think the system cannot identifier the correct digits.
The Overview shows incorrect data:
![image](https://user-images.githubusercontent.com/120333821/213446011-7a8f278c-ef70-42f3-b333-19318a623654.png)
![image](https://user-images.githubusercontent.com/120333821/213446396-5088031b-65cc-4c37-8ffd-89af07a2ced2.png)

Thanks for your help, and greetings from Germany.

**Comment by 8chM** [comment]
Tried a new setup, but datas are not correct.
![image](https://user-images.githubusercontent.com/120333821/213501268-b7fa5139-0b16-4dfa-bc36-ba21254d86b8.png)

What can I do?

**Comment by jomjol** [comment]
Two things to improve:
1) Choose a model, that is trained for digit only detection. The naming starts with: `dig-class11_...`
It has a much broader types of numbers trained, but no subdigits, which are not valid for LCD types anyway
2) Adjust your ROIs according to the documentation. They are too small. See: https://jomjol.github.io/AI-on-the-edge-device-docs/inital-setup/#detail-for-roi-configuration-digital-meters

**Reply by 8chM** [reply]
Die Klasse dig-class11 wurde verwendet, und im neuen Setup sind die Digits auch größer. Das Gerät ist noch nicht verbaut... Die Zahlen kamen von einem Foto.

Ich probiere es nochmal von vorne, und mach größere ROIs...
Gibt es sowas wie ein "Factory Reset", damit ich wieder ganz von vorne anfangen kann?

**Comment by jomjol** [comment]
Bei dig-class11 dürften eigentlich keine Nachkommastellen auftauchen:
![grafik](https://user-images.githubusercontent.com/30766535/213541068-ea2bae9e-68dd-401f-a6e8-31855b5939b9.png)
oder zumindest müssten sie alle ".0" sein.

**Comment by 8chM** [comment]
Ich habe das ganze neu aufgebaut... Testweise habe ich versucht, ob der Punkt erkannt wird, was wohl nicht der Fall ist, aber auch nicht schlimm.....
Ich habe Dig11 aber habe trotzdem habe ich Nachkommastellen.
![image](https://user-images.githubusercontent.com/120333821/213688163-fc06679e-8cf0-49e5-8067-0ec80f8cd7a0.png)

Was ich auch komisch finde, die ROIS sind eigentlich größer, wie sie bei der Recognition erkannt werden. Und wenn ich mir das ganze in der Overview anschaue, da sind scheinbar ein paar Werte richtig erkannt und erfasst worden.
![image](https://user-images.githubusercontent.com/120333821/213688449-3e5a6be1-450f-45c6-878e-4351278b2b96.png)
Was ich auch komisch finde, warum ist das Bild etwas schräg? Ich habe die Rotation nicht verändert, und die Installation ist fest und bewegt sich nicht. Das Bild ist ein Handyfoto und verändert sich auch nicht...

**Comment by 8chM** [comment]
Ok, ich habe nochmal bißchen rumgespielt, und jetzt passt es fast.... Vielen Dank bis hier her.
![image](https://user-images.githubusercontent.com/120333821/213696199-0caca9e7-6fe4-4421-81f0-3ce9178842c0.png)

Allerdings kommt es hier immer wieder zu Verschiebungen, das z.B. das Bild schräg dargestellt wird, obwohl es sich definitiv nicht verändert hat, und dann werden auch mal Werte nicht erkannt.
![image](https://user-images.githubusercontent.com/120333821/213696532-e95a993e-f860-43cd-951b-f0b3a8f507f9.png)

Bild Qualität habe ich mittlerweile auf 9 gestellt.

**Comment by jomjol** [comment]
Schräge Bilder kommen daher, dass die Referenz nicht erkannt wird. Am besten einen weißen Sticker mit schwarzem Punkt o.ä. aufkleben. Die Referenzen auf dem Bild haben grenzwertig wenig Kontrast.

**Comment by 8chM** [comment]
Danke dir, es funktioniert jetzt größtenteils:
![image](https://user-images.githubusercontent.com/120333821/214323048-f406602d-b8a5-4ac3-8265-c7a52805f9e7.png)

Was cool wäre, wenn man eine Zoom-Funktion einbauen könnte, zumindestens für die Positionierung der ROIs....

**Comment by 8chM** [comment]
Hättest du noch ein Paar Tipps, wie ich das After Processing einstelle, das da keine Phantasiewerte raus kommen?

pH: normalerweise bei 7.1 - kann zwischen 6.5 und 7.8 schwanken
Redox: normalerweise 650 - kann zwischen 400 - 800 schwanken
Temperatur: normalerweise 31 - kann zwischen 29 und kurzzeitig bei 34 liegen.

Ich habe zuerst gedacht, dass ich die MaxRateValue z.B für PH auf 0.7 einstelle, da mit ist er aber nicht klar gekommen. Daher habe ich vermutet, dass das Komma nicht einbezogen wird, und ich 70 einstellen muss. Liege ich da richtig?

Hier meine derzeitigen Settings:
![image](https://user-images.githubusercontent.com/120333821/214537280-c957ef47-94a4-4921-8d09-f6edc2b28fdb.png)
![image](https://user-images.githubusercontent.com/120333821/214537440-cedf7b8a-ad1c-459e-b08f-8d904e9b59ce.png)
![image](https://user-images.githubusercontent.com/120333821/214537511-83539047-23a0-4be8-a45e-33d42db25d26.png)
